packages/ics721/src/types.rs

```diff
@@ -0,0 +1,68 @@
+use cosmwasm_schema::{cw_serde, schemars::JsonSchema};
+use cosmwasm_std::Binary;
+use serde::{Deserialize, Serialize};
+
+use crate::{ibc::NonFungibleTokenPacketData, token_types::ClassId};
+
+#[derive(Serialize, Deserialize, JsonSchema, Clone, Debug, PartialEq)]
+#[allow(clippy::derive_partial_eq_without_eq)]
+#[schemars(crate = "cosmwasm_schema::schemars")]
+#[serde(crate = "cosmwasm_schema::serde")]
+pub struct Ics721Memo {
+    pub callbacks: Option<Ics721Callbacks>,
+}
+
+/// The format we expect for the memo field on a send
+#[cw_serde]
+pub struct Ics721Callbacks {
+    /// Data to pass with a callback on source side (status update)
+    /// Note - If this field is empty, no callback will be sent
+    pub src_callback_msg: Option<Binary>,
+    /// The address that will receive the callback message
+    pub src_msg_receiver: Option<String>,
+    /// Data to pass with a callback on the destination side (ReceiveNftIcs721)
+    /// Note - If this field is empty, no callback will be sent
+    pub dest_callback_msg: Option<Binary>,
+    /// The address that will receive the callback message
+    pub dest_msg_receiver: Option<String>,
```

Comment:
These 2 correlate with each other:
`src_callback_msg` and `src_msg_receiver`
`dest_callback_msg` and `dest_msg_receiver`

Maybe define an extra struct like `Callback` with binary and string props.

So it may looks like this:

```rs
pub struct Ics721Callbacks {

pub ack_callback: Option<Callback>,
pub receive_callback: Option<Callback>,
}
```

Reply:
Besides: in IBC specs it has been noted that the terms `source`, `target` resp `destination` are a bit misleading. Imo ack and receive may be better.

Reply:
Honestly, I get your point, but I really don't want to over complicate it, as the memo field is already some what complicated on its own.
```
{
  packet: {
    ..., 
    memo: {
      callbacks: {...}
    }
  }
}
```

Don't see any meaningful reason to complicate it even more.

Reply:
For me when using this in a contract, I was confused, with msg + address. So what may be complicated for you - was simpler for me - but fine with that :P.

But naming-wise this may help:
- src_callback_msg > ack_callback_data
- src_msg_receiver > ack_callback_addr (addr instead of receiver, otherwise it is confusing for receive callback below)
- dest_callback_msg > receive_callback_data
- dest_msg_receiver > receive_callback_addr